src/Surfnet/StepupMiddleware/ApiBundle/Controller/AuditLogController.php

```diff
@@ -31,11 +32,16 @@ public function secondFactorAuditLogAction(Request $request, Institution $instit
     {
         $this->denyAccessUnlessGranted(['ROLE_RA']);
 
+        $identityId = $request->get('identityId');
+        if (empty($identityId)) {
+            throw new BadApiRequestException(['This API-call MUST include the identityId as get parameter']);
```

Comment:
Out of curiosity as to why this fix would be needed I wanted to check something for myself :wink: If I remember correctly, required parameters are (or rather, should) either be tackled on request parsing (paramconverter, as is done for `Institution`) or as a requirement in the URL (see various routes). It seems this was intended to be the latter:
the route [has the requirement][1], but that is not [correctly][2] implemented, and the parameter itself is missing, in the [bundle][3] as well... And it seems the identity [is indeed required][4].

Perhaps adapting the route to enforce it through an explicit routing parameter (`/audit-log/{identityId}/second-factors`) would be better in the long run? Thoughts @relaxnow @rjkip ?

(if so, a tech debt ticket would suffice i.m.o, especially since it is in internal API)

[1]: https://github.com/SURFnet/Stepup-Middleware/blob/16550f3852f98a9e7d36c5b3cfdc20d7565abc72/src/Surfnet/StepupMiddleware/ApiBundle/Resources/config/routing.yml#L95-L96
[2]: https://github.com/SURFnet/Stepup-Middleware/blob/16550f3852f98a9e7d36c5b3cfdc20d7565abc72/src/Surfnet/StepupMiddleware/ApiBundle/Resources/config/routing.yml#L79-L80
[3]: https://github.com/SURFnet/Stepup-Middleware-clientbundle/blob/6974c91e3e84b680f569680167e6b670a639ae54/src/Surfnet/StepupMiddlewareClient/Identity/Service/AuditLogService.php#L54
[4]: https://github.com/SURFnet/Stepup-Middleware/blob/16550f3852f98a9e7d36c5b3cfdc20d7565abc72/src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/AuditLogRepository.php#L65